### PR TITLE
Soycode nodewrtc

### PR DESCRIPTION
**Pull request by soycode**
Includes basic rtcpeerconnection and rtcdatachannel support. Passes all but last integration test (onClose, which it fails with a timeout - I've done some investigation and suspect this may need fixing upstream).

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -40,6 +40,7 @@
     "tls-connect": "^0.2.2"
   },
   "optionalDependencies": {
+    "wrtc": "0.0.59",
```
Comment (willscott) [inline package.json]: is this stable? should it still be optional or move to normal?

**Comment by willscott** [comment]
check that CI isn't actually reporting something bad, otherwise looks good.

File: package.json (hunk 1 above)
Comment (soycode) [inline package.json]: It's optional because it breaks the build on OSX. I'm hoping to fix that upstream, though.

Comment (soycode) [inline package.json]: To be clear, "breaks the build" in this case means "npm install dies", not that any of our grunt processes fail. node-wrtc just doesn't compile nicely under OSX - they have related issues filed in their own repo.

**Comment by soycode** [comment]
Looks like Travis doesn't like installing node-wrtc, hence the failure. Shippable is a bit more annoying - jasmine_node wanted webrtc-adapter installed (even though it isn't loaded by node when actually run), and then chokes on some of its syntax. I'll see if I can figure out a better solution, ideally a way of convincing jasmine_node that it really doesn't need that module.

Locally, the tests all pass when the corresponding freedom build is symlinked in (except for the very last rtcpeerconnection integration test as noted).

**Comment by soycode** [comment]
So after a bit more testing/thinking, I believe that these CI failures are due to the dependency on the separate freedom branch that is different from the current npm version. These tests will continue to fail until a new freedom release is cut, but I trust the local passing enough to push ahead.

**Pull request merged**